fdbserver/DataDistribution.actor.h

```diff
@@ -352,17 +423,80 @@ class PhysicalShardCollection : public ReferenceCounted<PhysicalShardCollection>
 	uint64_t generateNewPhysicalShardID(uint64_t debugID);
 
 	// PhysicalShard Metrics
-	void updatePhysicalShardMetricsByKeyRange(KeyRange keys,
-	                                          StorageMetrics const& newMetrics,
-	                                          StorageMetrics const& oldMetrics,
-	                                          bool initWithNewMetrics);
+	std::vector<uint64_t> updatePhysicalShardMetricsByKeyRange(KeyRange keys,
+	                                                           StorageMetrics const& newMetrics,
+	                                                           StorageMetrics const& oldMetrics,
+	                                                           bool initWithNewMetrics);
 	void reduceMetricsForMoveOut(uint64_t physicalShardID, StorageMetrics const& metrics);
 	void increaseMetricsForMoveIn(uint64_t physicalShardID, StorageMetrics const& metrics);
 
 	// to remove
 	void printTeamPhysicalShardsMapping(std::string);
 };
 
+enum DataMoveType { PHYSICAL_SHARD_MOVE, READ_RANGE_MOVE };
+class DDEventBuffer : public ReferenceCounted<DDEventBuffer> {
```

Comment:
Can this be combined with DataDistributionRuntimeMonitor, or be a member of `DataDistributionRuntimeMonitor`. Then we can use a single API for `enqueuing` a DDEvent, and handle them synchronously.